### ISSUE TITLE
K-Median/Mean heuristic for ProtoNN missing in Pytorch

### ISSUE THREAD

**Issue by metastableB** [microsoft/EdgeML#138]
- The median heuristic function [utils.medianHeuristic](https://github.com/microsoft/EdgeML/blob/master/tf/edgeml/utils.py#L12) is missing in the pytorch branch. Please copy it to `edgeml_pytorch.utils`.
- It is being called from [here](https://github.com/microsoft/EdgeML/blob/0b33f1d959b8312615fac9bc941a28a16ce2cf95/pytorch/examples/ProtoNN/helpermethods.py#L49) when you run the example. That is, if gamma is not provided, we will estimate it.
- As can be seen from the Notes [here](https://github.com/microsoft/EdgeML/blob/0b33f1d959b8312615fac9bc941a28a16ce2cf95/tf/edgeml/utils.py#L12) and the code [here](https://github.com/microsoft/EdgeML/blob/master/tf/edgeml/utils.py#L54), the implementation is unaware of class labels and hence per-class k-means is not there. This requires effort :/ .
